core/blockchain.go

```diff
@@ -1221,6 +1221,9 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 			rawdb.WriteBody(batch, block.Hash(), block.NumberU64(), block.Body())
 			rawdb.WriteReceipts(batch, block.Hash(), block.NumberU64(), receiptChain[i])
 			rawdb.WriteTxLookupEntries(batch, block)
+			for _, tx := range block.Transactions() {
+				rawdb.WriteTransactionMeta(batch, tx.Hash(), tx.GetMeta())
+			}
```

Comment:
Was going to suggest this be parallelized, but there should only be 1 tx per block. Could assert that here and always just access `block.Transactions()[0]` but might just make sense to leave this as is 👍

Reply:
Good thinking about parallelizing it if there were many transactions. I think that we are going to have to completely redo the way the database works eventually.